monai/losses/aucm_loss.py

```diff
@@ -0,0 +1,172 @@
+                - ``"sum"``: the output will be summed.
+            gpu: if True, use GPU to calculate. Defaults to False.
+
+
+        Raises:
+            ValueError: margin must be in [0, 1]
+            TypeError: When ``other_act`` is not an ``Optional[Callable]``.
+            ValueError: and When more than 1 of [``sigmoid=True``, ``softmax=True``, ``other_act is not None``].
+                Incompatible values.
+        """
+        super().__init__(reduction=LossReduction(reduction).value)
+
+        if margin < 0 or margin > 1:
+            raise ValueError("imratio must be between 0 and 1")
+        if other_act is not None and not callable(other_act):
+            raise TypeError(f"other_act must be None or callable but is {type(other_act).__name__}.")
+        if int(sigmoid) + int(softmax) + int(other_act is not None) > 1:
+            raise ValueError("Incompatible values: more than 1 of [sigmoid=True, softmax=True, other_act is not None].")
+
+        self.margin = margin
+
+        self.p = pos_weight
+        self.num_classes = num_classes
+        if self.p:
+            assert len(self.p) == self.num_classes, "imratio must be a list of length num_classes"
+        else:
+            self.p = [0] * self.num_classes
+
+        self.to_onehot_y = to_onehot_y
+        self.sigmoid = sigmoid
+        self.softmax = softmax
+        self.other_act = other_act
+
+        self.a = torch.zeros(self.num_classes, dtype=torch.float32, requires_grad=True)
+        self.b = torch.zeros(self.num_classes, dtype=torch.float32, requires_grad=True)
+        self.alpha = torch.zeros(self.num_classes, dtype=torch.float32, requires_grad=True)
+
+    def forward(self, y_pred: torch.Tensor, y_true: torch.Tensor, auto=True) -> torch.Tensor:
+        """
+        Args:
+            y_pred: predicted tensor of shape [batch_size, num_classes, ...]
+            y_true: ground truth tensor of shape [batch_size, num_classes, ...]
+            auto: if True, automatically calculate imbalance ratio. Defaults to True.
+
+        Raises:
+            ValueError: When y_pred and num_classes have different sizes.
+            ValueError: When y_true and num_classes have different sizes.
+            ValueError: When y_pred and y_true have different sizes.
+        """
+        if y_pred.shape[1] != self.num_classes:
+            raise ValueError(
+                f"y_pred.shape[1] must be {self.num_classes}, equal to num_classes ,but is {y_pred.shape[1]}"
+            )
+
+        if y_true.shape[1] != self.num_classes:
+            raise ValueError(
+                f"y_true.shape[1] must be {self.num_classes}, equal to num_classes ,but is {y_true.shape[1]}"
+            )
+
+        if self.sigmoid:
+            y_pred = torch.sigmoid(y_pred)
+
+        n_pred_ch = y_pred.shape[1]
+        if self.softmax:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, `softmax=True` ignored.")
```

Comment:
Isn't it better to raise an error here instead of a warning?
Considering that `softmax` is defaulted to `False`, it means that the user has explicitly set it to `True` and the the input argument is wrongly set, so it might be better to raise a ValueError. Does it make sense?